src/cmp.rs

```diff
@@ -0,0 +1,1188 @@
+                let (_, arg) = param_str.split_once('=').unwrap();
+                arg.to_string()
+            };
+            let max_bytes = match max_bytes.parse::<usize>() {
+                Ok(num) => num,
+                Err(e) if *e.kind() == std::num::IntErrorKind::PosOverflow => usize::MAX,
+                Err(_) => {
+                    return Err(format!(
+                        "{}: invalid --bytes value '{}'",
+                        executable_str, max_bytes
+                    ))
+                }
+            };
+            params.max_bytes = Some(max_bytes);
+            continue;
+        }
+        if param == "-i" || param_str.starts_with("--ignore-initial=") {
+            let skip_desc = if param == "-i" {
+                opts.next()
+                    .ok_or_else(|| usage_string(&executable_str))?
+                    .to_string_lossy()
+                    .to_string()
+            } else {
+                let (_, arg) = param_str.split_once('=').unwrap();
+                arg.to_string()
+            };
+            let (skip_a, skip_b) = if let Some((skip_a, skip_b)) = skip_desc.split_once(':') {
+                (
+                    parse_skip(skip_a, &skip_desc)?,
+                    parse_skip(skip_b, &skip_desc)?,
+                )
+            } else {
+                let skip = parse_skip(&skip_desc, &skip_desc)?;
+                (skip, skip)
+            };
+            params.skip_a = Some(skip_a);
+            params.skip_b = Some(skip_b);
+            continue;
+        }
+        if param == "-s" || param == "-q" || param == "--quiet" || param == "--silent" {
+            params.quiet = true;
+            continue;
+        }
+        if param == "--help" {
+            println!("{}", usage_string(&executable_str));
+            std::process::exit(0);
+        }
+        if param_str.starts_with('-') {
+            return Err(format!("Unknown option: {:?}", param));
+        }
+        if from.is_none() {
+            from = Some(param);
+        } else if to.is_none() {
+            to = Some(param);
+        } else if skip_pos1.is_none() {
+            skip_pos1 = Some(parse_skip(&param_str, &param_str)?);
+        } else if skip_pos2.is_none() {
+            skip_pos2 = Some(parse_skip(&param_str, &param_str)?);
+        } else {
+            return Err(usage_string(&executable_str));
+        }
+    }
+
+    // Do as GNU cmp, and completely disable printing if we are
+    // outputing to /dev/null.
+    #[cfg(not(target_os = "windows"))]
+    if is_stdout_dev_null() {
+        params.quiet = true;
+        params.verbose = false;
+        params.print_bytes = false;
+    }
+
+    if params.quiet && params.verbose {
+        return Err(format!(
+            "{}: options -l and -s are incompatible",
+            executable_str
+        ));
+    }
+
+    params.from = if let Some(from) = from {
+        from
+    } else if let Some(param) = opts.next() {
+        param
```

Comment:
Isn't this already handled at the end of the while loop above?

Reply:
Right, my mistake. My future self would appreciate a comment there because I know I might be caught again thinking that this is duplicated functionality. Given that this is already implemented like this in diff though, we can leave the addition of such a comment (in both places) for later.